kustomize/commands/localize/localize_test.go

```diff
@@ -122,10 +121,6 @@ func TestOutput(t *testing.T) {
 	successMsg := fmt.Sprintf(`SUCCESS: localized "%s" to directory %s
 `, target.String(), target.Join("dst"))
 	require.Equal(t, successMsg, buffy.String())
-
-	const msg = "Check that cmd log output is hooked to buffy."
-	log.Print(msg)
-	require.Contains(t, buffy.String(), msg)
```

Comment:
I believe what we need to do in this test is what was instead done in the production code: set `log.SetOutput` to the buffer. I see we use this technique in `TestResolveVarsOneUnused` for example

Reply:
Why do we check this?

Reply:
Yes, I was trying to test that log.SetOutput took effect. At the time of writing (I thought) all other calls to log in the implementation were for impossible cases aka cases that we should never realistically hit (I missed 1 error case). As a result, I had no way to test that localize was printing to writer other than printing to it myself.

I agree that we could use log.SetOutput to test the success/error messages. It would be cleaner than printing and searching through stderr in tests. We would be under the assumption that the implementation uses log to print to stderr, but I think that's ok?

I just found that tests in api/internal/localizer/locloader_test.go use log.SetOutput without reinstating stderr after test case completion. I can fix that (and anything else we find out of scope of this PR) in a follow-up.

Reply:
I can toss that fix in here since it's related!